### PR TITLE
Benji/ios 531 read notification

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: Jibber.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -197,8 +196,8 @@
 		D62C9CC3266C3CAC00BB9632 /* UserDefaultsManager.swift in Sources */ = {isa = PBXBuildFile; fileRef = D62C9CC0266C3CA800BB9632 /* UserDefaultsManager.swift */; };
 		D62D16322736F60400C542CE /* ToastErrorView.swift in Sources */ = {isa = PBXBuildFile; fileRef = D62D16312736F60400C542CE /* ToastErrorView.swift */; };
 		D62D16342736F60400C542CE /* ToastErrorView.swift in Sources */ = {isa = PBXBuildFile; fileRef = D62D16312736F60400C542CE /* ToastErrorView.swift */; };
-		D62ED579279CA2ED00F7C08A /* BorderedPersoniew.swift in Sources */ = {isa = PBXBuildFile; fileRef = D62ED578279CA2ED00F7C08A /* BorderedPersoniew.swift */; };
-		D62ED57A279CB4A400F7C08A /* BorderedPersoniew.swift in Sources */ = {isa = PBXBuildFile; fileRef = D62ED578279CA2ED00F7C08A /* BorderedPersoniew.swift */; };
+		D62ED579279CA2ED00F7C08A /* BorderedPersonView.swift in Sources */ = {isa = PBXBuildFile; fileRef = D62ED578279CA2ED00F7C08A /* BorderedPersonView.swift */; };
+		D62ED57A279CB4A400F7C08A /* BorderedPersonView.swift in Sources */ = {isa = PBXBuildFile; fileRef = D62ED578279CA2ED00F7C08A /* BorderedPersonView.swift */; };
```
Comment (Benjamindodgson) [inline Jibber.xcodeproj/project.pbxproj:200]: Naming fix

File: Shared/Extensions/String+Extensions.swift [hunk 2]
```diff
@@ -66,8 +66,19 @@ extension String {
     func trimWhitespace() -> String {
         return trimmingCharacters(in: .whitespacesAndNewlines)
     }
+    
+    var isValidGivenName: Bool {
```
Comment (Benjamindodgson) [inline Shared/Extensions/String+Extensions.swift:70]: Shows different messages based on validation

File: iOS/Extensions/CNContact+Extensions.swift [hunk 3]
```diff
@@ -111,6 +111,7 @@ extension CNContact: PersonType {
 
         // Set the font size to something large so it will assured to be shrunk down to fit the label.
         label.font = FontType.regularBold.font.withSize(100)
+        label.textColor = ThemeColor.T1.color
```
Comment (Benjamindodgson) [inline iOS/Extensions/CNContact+Extensions.swift:114]: Fixes label on light mode

File: iOS/Flows/People/PeopleCoordinator.swift [hunk 4]
```diff
@@ -223,14 +227,18 @@ extension PeopleCoordinator: MFMessageComposeViewControllerDelegate {
...
+    
     func messageComposeViewController(_ controller: MFMessageComposeViewController,
                                       didFinishWith result: MessageComposeResult) {
         switch result {
         case .cancelled, .failed:
             controller.dismiss(animated: true)
         case .sent:
-            AnalyticsManager.shared.trackEvent(type: .inviteSent, properties: nil)
+            var properties: [String: Any] = [:]
+            if let rsvp = self.selectedReservation?.objectId {
```
Comment (Benjamindodgson) [inline iOS/Flows/People/PeopleCoordinator.swift:238]: Allows us to associate an invite to what that invited user does.

File: iOS/Flows/Profile/Context Cues/ContextCueCollectionViewLayout.swift [hunk 5]
```diff
@@ -20,14 +20,14 @@ class ContextCueCollectionViewLayout: UICollectionViewCompositionalLayout {
             switch sectionType {
             case .contextCues:
                 
-                //let inset = Theme.ContentOffset.xtraLong.value
+                let inset = environment.container.contentSize.width.half
```
Comment (Benjamindodgson) [inline iOS/Flows/Profile/Context Cues/ContextCueCollectionViewLayout.swift:23]: Fixes offsets.

File: iOS/Flows/Profile/Context Cues/ContextCuesViewController.swift [hunk 6]
```diff
@@ -31,6 +32,10 @@ class ContextCuesViewController: DiffableCollectionViewController<ContextCueColl
         self.view.insertSubview(self.lineView, belowSubview: self.collectionView)
         self.lineView.set(backgroundColor: .B2)
         self.lineView.alpha = 0.5
+        
+        if let user = self.person as? User, user.isCurrentUser {
```
Comment (Benjamindodgson) [inline iOS/Flows/Profile/Context Cues/ContextCuesViewController.swift:36]: Pins the add above the collection view.

File: iOS/Flows/Profile/Context Cues/Creator/ContextCueCreatorViewController.swift [hunk 7]
```diff
@@ -99,9 +99,14 @@ class ContextCueCreatorViewController: DiffableCollectionViewController<EmojiCol
         super.viewDidLayoutSubviews()
         
         self.segmentControl.sizeToFit()
-        self.segmentControl.centerOnX()
         self.segmentControl.pinToSafeAreaTop()
         
+        let segmentWidth = (self.view.width - Theme.ContentOffset.standard.value.doubled) * 0.125
+        for i in 0...self.segmentControl.numberOfSegments - 1 {
```
Comment (Benjamindodgson) [inline iOS/Flows/Profile/Context Cues/Creator/ContextCueCreatorViewController.swift:105]: Fixes spacing.

File: iOS/_Stream Temp/ChatClient+Extensions.swift [hunk 8]
```diff
@@ -121,7 +121,7 @@ extension ChatClient {
         return self.messageController(cid: msg.cid!, messageId: msg.id)
     }
 
-    func message(cid: ConversationId, id: MessageId) -> Message {
-        return self.messageController(cid: cid, messageId: id).message!
+    func message(cid: ConversationId, id: MessageId) -> Message? {
```
Comment (Benjamindodgson) [inline iOS/_Stream Temp/ChatClient+Extensions.swift:124]: This was causing a crash for some reason.

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**